scopes/dependencies/modules/packages-excluder/generate-node-modules-pattern.spec.ts

```diff
@@ -0,0 +1,52 @@
+import { generateNodeModulesPattern } from './generate-node-modules-pattern';
+
+describe('generateNodeModulesPattern()', () => {
+  it('should return an empty regex when the received array is empty', () => {
+    expect(generateNodeModulesPattern({ packages: [] })).toEqual('node_modules/(?!()/)');
+  });
+  it('should work with more than one packages', () => {
```

Comment:
you need to better restructure your tests in groups, and separate the use cases. here is some pseudo-code
- describe (should work with more than one packages)
  - before - create the pattern and the regex
  - describe (should exclude first package)
    - it (should have yarn structure excluded)
    - it (should have old pnpm structure excluded)
    - it (should have new pnpm structure excluded)
  - describe (should exclude the second package)
    - it (should have yarn structure excluded)
    - it (should have old pnpm structure excluded)
    - it (should have new pnpm structure excluded)

Reply:
done